File: client/src/models/actor/npc/NonPlayerActor.ts
import Actor, {ActorSkill} from "../Actor";
import Ability from "../../Ability";
import { ActorTalent } from "../../Talent";
import Injury from "../../Injury";

export enum RatingType {
    Combat = 'Combat',
    Social = 'Social',
    General = 'General'
}

export default interface NonPlayerActor extends Actor {
    combat: number,
    social: number,
    general: number,
    abilities: Ability[],
    talents: ActorTalent[]
}

export interface SingleNonPlayerCharacter extends NonPlayerActor {
    skills: ActorSkill[]
    injuries: Injury[]
}

export const getRatings = (npc: NonPlayerActor): string => {
    return '[combat] ' + String(npc?.combat!!) + ' [social] ' + String(npc?.social!!) + ' [general] ' + String(npc?.general!!)
}

export const combat = '[combat]'
export const social = '[social]'
export const general = '[general]'

File: server/src/main/java/com/github/genraven/genesys/domain/actor/npc/NonPlayerActor.java
package com.github.genraven.genesys.domain.actor.npc;

import com.github.genraven.genesys.domain.Ability;
import com.github.genraven.genesys.domain.actor.Actor;
import lombok.Data;
import lombok.EqualsAndHashCode;

import java.util.ArrayList;
import java.util.List;

@EqualsAndHashCode(callSuper = true)
@Data
public class NonPlayerActor extends Actor {

    protected NonPlayerActor() {
    }

    public NonPlayerActor(final Actor actor) {
        this.setName(actor.getName());
        this.setBrawn(actor.getBrawn());
        this.setAgility(actor.getAgility());
        this.setIntellect(actor.getIntellect());
        this.setCunning(actor.getCunning());
        this.setWillpower(actor.getWillpower());
        this.setPresence(actor.getPresence());
        this.setWounds(actor.getWounds());
        this.setSoak(actor.getSoak());
        this.setWeapons(actor.getWeapons());
        this.setArmors(actor.getArmors());
    }

    private int combat = 1;
    private int social = 1;
    private int general = 1;
    private List<Ability> abilities = new ArrayList<>();
}
